### PR TITLE
Fix the issues of utils_netperf

### PR DESCRIPTION
**Pull request by luckyh**
* correct the package remove command
* explicitly close sessions to avoid issues

Signed-off-by: Xu Han <xuhan@redhat.com>

### CODE REVIEW

**Comment by yanglei-rh** [comment]
LGTM
The current code solves the problem encountered when testing netperf in the past: `[Errno 9] Bad file descriptor.` Use the current code to test, in the entire 8.5 test, this problem has not been reproduced, proving that the current code solves the past the problems encountered.

**Review by yanglei-rh — APPROVED** [review]

File: virttest/utils_netperf.py [hunk 1]
```diff
@@ -104,21 +104,39 @@ def __init__(self, address, netperf_path, md5sum="", netperf_source="",
...
+            return
+        self._session.close()
+        self._session = None
 
     def env_cleanup(self, clean_all=True):
         clean_cmd = ""
         if self.netperf_dir:
             clean_cmd = "rm -rf %s" % self.netperf_dir
         if clean_all:
-            clean_cmd += " rm -rf %s" % os.path.join(self.remote_path,
```
Comment (PaulYuuu) [inline virttest/utils_netperf.py:119]: Nice catch.

**Review by PaulYuuu — APPROVED** [review]

File: virttest/utils_netperf.py (hunk 1 above)
Comment (luckyh) [reply virttest/utils_netperf.py:119]: Actually the current change is still not correct, I'm working on a new version and will send it out soon, thanks.

**Comment by luckyh** [comment]
@yanglei-rh @PaulYuuu @yanan-fu @leidwang would you please help review this patch? thanks!

**Comment by yanglei-rh** [comment]
Hello @luckyh  I tried to use this patch to test netperf stress test, but test failed. After checking the test log, I found that the executed command was wrong:` "start /b c:\\/netserver-2.6.0.exe > null"`. The correct command should be: "start /b c:\\netserver-2.6.0.exe > null". There is an extra "/" before netserver-2.6.0.exe. Could you please help check it again? Thanks a lot.

**Comment by luckyh** [comment]
> Hello @luckyh I tried to use this patch to test netperf stress test, but test failed. After checking the test log, I found that the executed command was wrong:` "start /b c:\\/netserver-2.6.0.exe > null"`. The correct command should be: "start /b c:\netserver-2.6.0.exe > null". There is an extra "/" before netserver-2.6.0.exe. Could you please help check it again? Thanks a lot.

@yanglei-rh hmm, that's interesting, could you please try reverting this patch to see whether the netperf command line would be changed in some way? thanks!

**Comment by yanglei-rh** [comment]
> > Hello @luckyh I tried to use this patch to test netperf stress test, but test failed. After checking the test log, I found that the executed command was wrong:` "start /b c:\\/netserver-2.6.0.exe > null"`. The correct command should be: "start /b c:\netserver-2.6.0.exe > null". There is an extra "/" before netserver-2.6.0.exe. Could you please help check it again? Thanks a lot.
> 
> @yanglei-rh hmm, that's interesting, could you please try reverting this patch to see whether the netperf command line would be changed in some way? thanks!
Sorry, it's my mistake,I reversed the two results. Please review it again:
Before applying the patch(test pass):`'start /b c:\\/netserver-2.6.0.exe > null'`
After applying the patch(test failed):`'start /b c:\\netserver-2.6.0.exe > null'`

File: virttest/utils_netperf.py [hunk 2]
```diff
@@ -99,26 +111,45 @@ def __init__(self, address, netperf_path, md5sum="", netperf_source="",
...
-                                             self.netperf_file)
-
-        LOG.debug("Create remote session")
-        self.session = remote.remote_login(self.client, self.address,
-                                           self.port, self.username,
-                                           self.password, prompt,
-                                           linesep, timeout=360,
-                                           status_test_command=status_test_command)
+            self.netserver_path = self.__remote_file
+            self.netperf_path = self.__remote_file
```
Comment (yanglei-rh) [inline virttest/utils_netperf.py:115]: Hello @luckyh It should be the problem caused by this change.

Comment (luckyh) [reply virttest/utils_netperf.py:115]: ok, could you please let me know about the error message you got? thanks!

Comment (yanglei-rh) [reply virttest/utils_netperf.py:115]: > ok, could you please let me know about the error message you got? thanks!

Hello @luckyh I have update it in bugzilla attachment.

Comment (luckyh) [reply virttest/utils_netperf.py:115]: Have updated the patch, thanks!

**Comment by yanglei-rh** [comment]
Test pass
(1/1) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.netperf_stress_test.TCP_STREAM.guest2guest.q35: PASS (360.08 s)

**Comment by leidwang** [comment]
Test passed with this patch.
JOB LOG    : /root/avocado/job-results/job-2022-04-11T04.44-fb3825d/job.log
 (1/1) Host_RHEL.m.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.netperf_stress_test.TCP_STREAM.guest2guest.q35: STARTED
\ (1/1) Host_RHEL.m.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.netperf_stress_test.TCP_STREAM.guest2guest.q35: PASS (358.39 s)

File: virttest/utils_netperf.py [hunk 3]
```diff
@@ -273,6 +307,13 @@ def __init__(self, address, netperf_path, md5sum="", netperf_source="",
                                             password, prompt, linesep, status_test_command,
                                             compile_option, install)
 
+    @property
+    def netserver_bin(self):
+        get_basename = os.path.basename
+        if self.client == "nc":
+            get_basename = ntpath.basename
+        return get_basename(self.netserver_path)
+
```
Comment (PaulYuuu) [inline virttest/utils_netperf.py:316]: @luckyh, since these lines and L374~L380 are dupped, can we merge them in the parent class `Netperf`?

Comment (PaulYuuu) [reply virttest/utils_netperf.py:316]: Sorry, they are similar but not the same, so this is good.

**Review by PaulYuuu — APPROVED** [review]

**Comment by yanan-fu** [comment]
Code LGTM, and regression test pass in my side.

Ack

**Comment by luckyh** [comment]
Thanks to all.

**Pull request merged**